File: Implementações/C/2ou1.c
#include <stdio.h>
#include <stdbool.h>

int main() {
    int andre, carlos, pietro;
    bool jogadaValida;

    puts("Doooooois ou um!!\n");

//Primeiro jogador

    do
    {
        puts("Digite o valor 1 ou 2 para Andre: ");
        scanf("%d", &andre);
        jogadaValida = true;
        
        if (andre != 1 && andre != 2) {
            jogadaValida = false;
            puts("Digite apenas 1 ou 2!!\n");
        }

    } while (jogadaValida != true);
    
//Segundo Jogador

    do
    {
        puts("Digite o valor 1 ou 2 para Carlos: ");
        scanf("%d", &carlos);
        jogadaValida = true;

        if (carlos != 1 && carlos != 2) {
            jogadaValida = false;
            puts("Digite apenas 1 ou 2!!\n");
        } 

    } while (jogadaValida != true);

//terceiro jogador:

    do
    {
        puts("Digite o valor 1 ou 2 para Pietro: ");
        scanf("%d", &pietro);
        jogadaValida = true;

        if (pietro != 1 && pietro != 2) {
            jogadaValida = false;
            puts("Digite apenas 1 ou 2!!\n");
        } 

    } while (jogadaValida != true);

//Faz as conparações e imprime o vencedor:

    if (andre == 1 && carlos == 1 && pietro == 2) {
        puts("Pietro é o vencedor!!");    
        }
        else if (andre == 2 && carlos == 2 && pietro == 1 ) {
            puts("Pietro é o vencedor!!");
            }
            else if (andre == 1 && carlos == 2 && pietro == 1)
               {
                puts("Carlos é o vencedor!!");
                }
                 else if (andre == 2 && carlos == 1 && pietro == 2){
                    puts("Carlos é o vencedor!!");
                    }
                      else if (andre == 1 && carlos == 2 && pietro == 2){
                        puts("Andre é o vencedor!!");
                        }
                         else if (andre == 2 && carlos == 1 && pietro == 1){
                         	puts("Andre é o vencedor!!");
                        	}
                         	else {
								puts("Empatooooouuuu!!!");
								}

    return 0;
}

File: Implementações/rust/2ou1.rs
use std::io;

fn main() {
    let mut andre;
    let mut carlos;
    let mut pietro;

    println!("Dois ou Um!!");
      
    loop {
        
        println!("Digite 2 ou 1 para Andre: ");
//Cria uma nova string, lê a entrada de dados e armazena na variável 'andre'
        andre = String::new();
        io::stdin().read_line(&mut andre).expect("Erro ao criar string.");
//converte a string para número inteiro:
        let andre: i32 = andre.trim().parse().ok().expect("Digite apenas 2 ou 1, não uilize ponto ou virgula");
//Condição para quebrar o loop quando retornar verdadeiro
        if andre == 2 || andre == 1 {
        break;
        }
    }

//Segundo jogador: 
    loop {
            
        println!("Digite 2 ou 1 para Carlos: ");
        carlos = String::new();
        io::stdin().read_line(&mut carlos).expect("Erro ao criar string.");

        let carlos: i32 = carlos.trim().parse().ok().expect("Digite apenas 2 ou 1, não uilize ponto ou virgula");

        if carlos == 2 || carlos == 1 {
            break;
            }
    }

//terceiro jogador:
    loop {
        println!("Digite 2 ou 1 para Pietro: ");
        pietro = String::new();
        io::stdin().read_line(&mut pietro).expect("Erro ao criar string.");
    
        let pietro: i32 = pietro.trim().parse().ok().expect("Digite apenas 2 ou 1, não uilize ponto ou virgula");

        if pietro == 2 || pietro == 1 {
            break;
            }
    }

//comparações:
    if andre == carlos && carlos == pietro {
        println!("Empatooou!!");
    } 
        else if andre == carlos {
            println!("Pietro é o vencedor!!");
        } 
            else if pietro == andre {
                println!("Carlos  é o vencedor!!")
            } 
                else {
                    println!("André é o vencedor!!");
                }
    }
